github-rag/app.py

```
@@ -13,51 +15,117 @@
     """Extract repository name from URL"""
     try:
-        return url.split('/')[-1].replace('.git', '')
+        parts = url.split('/')
+        if len(parts) < 5:
+            raise ValidationError("Invalid repository URL format")
+        repo_name = parts[-1].replace('.git', '')
+        if not repo_name:
+            raise ValidationError("Could not extract repository name from URL")
+        return repo_name
+    except Exception as e:
+        raise ValidationError(f"Failed to extract repository name: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)